octo/run-tests.sh

```diff
@@ -2,4 +2,4 @@
 
 set -euo pipefail
 
-richgo test ./...
+richgo test -short ./...
```

Comment:
Can you expand on this change?

I've not seen `-short` used before but a quick read of the docs and it looks to be helpful if we are using `testing.Short()` in our actual tests. That's not something we're using though. Even if we were, I think I would want to have all unit tests run in CI though.

Is there another benefit to adding this? or is this part of how you're gating the integration tests? Just trying to understand. Thanks

Reply:
Gotcha. I think I'm OK with that.

What about something like is being done with some of the other paketo buildpacks:  https://github.com/paketo-buildpacks/go-dist/blob/main/scripts/integration.sh#L124

```
GOMAXPROCS="${GOMAXPROCS:-4}" go test -count=1 -timeout 0 ./integration/... -v -run Integration
```

This would only run the integration tests. The issue would be with our present unit tests scripts which would run everything, including integration tests, but perhaps we could emit different unit test scripts for projects that have integration tests and projects that don't.

i.e. If you add integration tests, then you're going to need to modify your unit tests. If you have no integration tests, then it's OK for the unit tests to just run everything.

I'm open to either solution, but if adding integration tests means that you need to make changes to the project with the integration tests. I'm OK with that. I just don't want to add something here and have to go change all of our projects w/out integration tests.

Reply:
@dmikusa-pivotal, I think I've got a working example here, it seems to work locally quite well. If there are no integration tests then nothing is changed.  If they exist, then the unit tests are configured with the `-short` flag, and the integration test pipeline is added (only running `./integration/...`) which is I think what we're after.

I've also added a unit test so I could see what was happening, I was surprised to see that there are no other tests in this repo - I've added it as a standard go test, rather than a ginkgo/bdd style test - happy to change.

Reply:
I've run this manually against a PR on the apache-tomee repo to see what it looks like - https://github.com/garethjevans/apache-tomee/pull/55/commits/0d7f8236b17430e8f0116d5368ad95ab3a7f8854